src/Symfony/Component/Lock/Lock.php

```diff
@@ -36,13 +36,21 @@ final class Lock implements LockInterface, LoggerAwareInterface
     private $dirty = false;
 
     /**
-     * @param Key            $key         Resource to lock
-     * @param StoreInterface $store       Store used to handle lock persistence
-     * @param float|null     $ttl         Maximum expected lock duration in seconds
-     * @param bool           $autoRelease Whether to automatically release the lock or not when the lock instance is destroyed
+     * @param Key                   $key         Resource to lock
+     * @param PersistStoreInterface $store       Store used to handle lock persistence
+     * @param float|null            $ttl         Maximum expected lock duration in seconds
+     * @param bool                  $autoRelease Whether to automatically release the lock or not when the lock instance is destroyed
      */
-    public function __construct(Key $key, StoreInterface $store, float $ttl = null, bool $autoRelease = true)
+    public function __construct(Key $key, $store, float $ttl = null, bool $autoRelease = true)
     {
+        if ($store instanceof StoreInterface) {
+            @trigger_error(sprintf('"%s" is deprecated since Symfony 4.4 and has been splitted into "%s", "%s", "%s".', StoreInterface::class, PersistStoreInterface::class, ExpiringStoreInterface::class, BlockingStoreInterface::class), E_USER_DEPRECATED);
```

Comment:
this should not be triggered here. The deprecation of StoreInterface itself will be reported by DebugClassLoader when you implement it.